libs/core/src/main/java/org/opensearch/core/xcontent/MediaTypeParserRegistry.java

```diff
@@ -48,7 +55,25 @@ public final class MediaTypeParserRegistry {
     // Default mediaType singleton
     private static MediaType DEFAULT_MEDIA_TYPE;
 
-    public static void register(MediaType[] acceptedMediaTypes, Map<String, MediaType> additionalMediaTypes) {
+    // JSON is a core type, so we create a static instance for implementations that require JSON format (e.g., tests)
+    // todo we should explore moving the concrete JSON implementation from the xcontent library to core
+    public static final MediaType JSON;
+
+    static {
+        List<MediaType> mediaTypes = new ArrayList<>();
+        Map<String, MediaType> amt = new HashMap<>();
+        for (MediaTypeProvider provider : ServiceLoader.load(MediaTypeProvider.class)) {
```

Comment:
We probably need to be careful with classloaders here:
```suggestion
        for (MediaTypeProvider provider : ServiceLoader.load(MediaTypeProvider.class, MediaTypeProvider.class.getClassLoader())) {
```
(PS: We could also use NamedSPILoader from Apache Lucene, as an option)

Reply:
I think passing in the classloader here is fine. It's not done for any other SPI interfaces though (e.g., `NamedXContentProvider`, `XContentBuilderExtension`) so I don't think it's a blocker, for now. Could be a good idea though if we start writing custom classloaders for security purposes as we isolate the API logic. I'd rather not use lucene's NamedSPILoader though because the focus of that is for lucene specific Codecs, Postings, etc.